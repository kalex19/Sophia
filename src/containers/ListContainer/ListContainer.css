.list-container{
  margin: 5 30%;
}