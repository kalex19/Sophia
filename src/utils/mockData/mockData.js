export const mockList = {
listId: 1,
title: 'Target',
};

export const mockItem = {
itemId: 1,
listId: 1,
isComplete: false,
isImportant: false,
text: 'Purchase a birthday card and pink gift wrap',
complete_by_date: '7/14/19',
};

export const mockLoading = false;

export const mockError = "There is an error";