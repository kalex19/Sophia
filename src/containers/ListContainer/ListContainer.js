import React, { Component } from 'react';
import List from '../../containers/Lists/List';
import { connect } from 'react-redux';
import { getItems, getLists } from '../../utils/apicalls';
import * as actions from '../../actions';
import './ListContainer.css';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

export class ListContainer extends Component {
	async componentDidMount () {
		this.fetchItems();
		this.fetchLists();
	}

	fetchItems = async () => {
		const { items } = this.props;
		if (!items.length) {
			try {
				const items = await getItems();
				this.props.addAllItems(items);
				this.props.hasError('');
			} catch (err) {
				this.props.hasError(err.message);
			}
		}
	};

	fetchLists = async () => {
		const { lists } = this.props;
		if (!lists.length) {
			try {
				const lists = await getLists();
				this.props.addAllLists(lists);
				this.props.hasError('');
			} catch (err) {
				this.props.hasError(err.message);
			}
		}
	};

	render () {
		const lists = this.props.lists.map(list => {
			const items = this.props.items.filter(item => item.list_id == list.id);
			return (
				<Link to={`/lists/${list.id}`} style={{textDecoration:'none'}}>
					<div>
						<h3 className="list-link">{list.title} ({items.length})</h3>
					</div>
				</Link>
			);
		});
		return <div className="list-container">{lists}</div>;
	}
}

export const mapStateToProps = state => ({
	items: state.items,
	lists: state.lists
});

export const mapDispatchToProps = dispatch => ({
	addAllItems: items => dispatch(actions.addAllItems(items)),
	addAllLists: lists => dispatch(actions.addAllLists(lists)),
	hasError: message => dispatch(actions.hasError(message))
});

export default connect(mapStateToProps, mapDispatchToProps)(ListContainer);

ListContainer.propTypes = {
	items: PropTypes.array,
	lists: PropTypes.array,
	addAllItems: PropTypes.func,
	addAllLists: PropTypes.func,
	hasError: PropTypes.func
};
