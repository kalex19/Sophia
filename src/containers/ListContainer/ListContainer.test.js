import {shallow} from 'enzyme';
import React from 'react';
import {ListContainer, mapStateToProps, mapDispatchToProps} from './ListContainer';
import {mockState, mockItems, mockLists} from '../../utils/mockData/mockData';

describe('ListContainer', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<ListContainer lists={mockLists} items={mockItems}/>)
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  });

  describe('CDM', () => {

  });

  describe('fetchItems', () => {

  });

  describe('fetchLists', () => {
    
  });

  describe('MSTP', () => {
  it('should return an array of items', () => {
    const mappedProps=mapStateToProps(mockState);
    expect(mappedProps).toEqual(mockState)
  })
  });

  describe('MDTP', () => {

  });
})