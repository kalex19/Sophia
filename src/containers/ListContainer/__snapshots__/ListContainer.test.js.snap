// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListContainer should match snapshot 1`] = `ShallowWrapper {}`;
